src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProvider.php

```diff
@@ -113,21 +123,47 @@ public function read(array $domains, array $locales): TranslatorBag
 
             foreach ($locales as $locale) {
                 if ($locale !== $this->defaultLocale) {
-                    $content = $this->exportProjectTranslations($locale, $fileId);
+                    $response = $this->exportProjectTranslations($locale, $fileId);
                 } else {
-                    $content = $this->downloadSourceFile($fileId);
+                    $response = $this->downloadSourceFile($fileId);
                 }
 
-                $translatorBag->addCatalogue($this->loader->load($content, $locale, $domain));
+                $responses[] = [
+                    'response' => $response,
+                    'locale' => $locale,
+                    'domain' => $domain,
+                ];
             }
         }
 
+        foreach ($responses as $responseData) {
+            /** @var ResponseInterface $response */
+            $response = $responseData['response'];
+
+            if (204 === $response->getStatusCode()) {
+                throw new ProviderException(sprintf('No content in exported file. Message: "%s".', $response->getContent(false)), $response);
+            }
+
+            if (200 !== $response->getStatusCode()) {
+                throw new ProviderException(sprintf('Unable to export file. Message: "%s".', $response->getContent(false)), $response);
+            }
+
+            $exportResponse = $this->filesDownloader->request('GET', $response->toArray()['data']['url']);
+
+            if (200 !== $exportResponse->getStatusCode()) {
+                throw new ProviderException(sprintf('Unable to download file content. Message: "%s".', $response->getContent(false)), $exportResponse);
```

Comment:
In this loop, we shouldn't throw any exception, but use the logger to avoid breaking the loop and lose the rest of the responses.

Reply:
Used logger